### PR TITLE
Added support for surviving worker count

### PR DESCRIPTION
**Pull request by ghost**
"I am dying, but there are X other workers still running."

Passing a tag name/value(s) will enable an additional query of
the AWS API to count similarly-tagged instances in the worker
pool. The count will be included in the notification message.

### CODE REVIEW

**Comment by osellers** [comment]
I'd be interested to hear people's thoughts on this addition. It's intended to give Ops users an immediate indication that all is well in their cluster(s) when they receive Slack notification about terminated spot instances, without them needing to look elsewhere or rely on additional monitoring.

**Comment by max-rocket-internet** [comment]
I get the idea but is the log output of this daemonset really the right place to "give Ops users an immediate indication"?

Wouldn't it make more sense to use a cloudwatch metric? e.g. if `GroupDesiredCapacity != GroupInServiceInstances` for longer than 5 mins then something has gone wrong with scaling

**Comment by osellers** [comment]
Hi @max-rocket-internet. Possibly not the right place, just _another_ place. It's something I found myself wanting when I saw spot termination notices, so wanted to write the code and put it out there in case others saw value in it. I agree with you that robust external monitoring of actual versus desired capacity should also be in place.

It makes perfect sense to me to limit the tasks of this handler to the minimum it should do.

File: entrypoint.sh [hunk 1]
```diff
@@ -66,8 +66,24 @@ while http_status=$(curl -o /dev/null -w '%{http_code}' -sL "${NOTICE_URL}"); [
...
+  SWC_COUNT=$(aws ec2 --cli-read-timeout $SWC_TIMEOUT --output text --query "Reservations[*].Instances[*] | length(@)" --region $REGION describe-instances --filters "Name=instance-state-name,Values=running" "Name=tag:$SWC_TAG_NAME,Values=$SWC_TAG_VALUES" 2>/dev/null)
+  case $SWC_COUNT in
+    ''|*[!0-9]*) SWC_MESSAGE="Failed" ;;
+    *) SWC_MESSAGE="$((SWC_COUNT-1))" ;;
+  esac
+fi
+
 echo "$(date): ${http_status}"
-MESSAGE="Spot Termination${CLUSTER_INFO}: ${NODE_NAME}, Instance: ${INSTANCE_ID}, Instance Type: ${INSTANCE_TYPE}, AZ: ${AZ}"
+MESSAGE="Spot Termination${CLUSTER_INFO}: ${NODE_NAME}, Instance: ${INSTANCE_ID}, Instance Type: ${INSTANCE_TYPE}, AZ: ${AZ}, Surviving Worker Count: ${SWC_MESSAGE}"
```
Comment (egeland) [inline entrypoint.sh]: It seems like even if the variables aren't set, you'd get `Surviving Worker Count:` in the output.
If we don't set variables, we shouldn't get that output, makes it messy.
Am I reading it correctly?

File: entrypoint.sh [hunk 2]
```diff
@@ -89,7 +105,7 @@ fi
 #
 if [ "${SLACK_URL}" != "" ]; then
   color="danger"
-  curl -X POST --data "payload={\"attachments\":[{\"fallback\":\"${MESSAGE}\",\"title\":\":warning: Spot Termination${CLUSTER_INFO}\",\"color\":\"${color}\",\"fields\":[{\"title\":\"Node\",\"value\":\"${NODE_NAME}\",\"short\":false},{\"title\":\"Instance\",\"value\":\"${INSTANCE_ID}\",\"short\":true},{\"title\":\"Instance Type\",\"value\":\"${INSTANCE_TYPE}\",\"short\":true},{\"title\":\"Availability Zone\",\"value\":\"${AZ}\",\"short\":true}]}]}" "${SLACK_URL}"
+  curl -X POST --data "payload={\"attachments\":[{\"fallback\":\"${MESSAGE}\",\"title\":\":warning: Spot Termination${CLUSTER_INFO}\",\"color\":\"${color}\",\"fields\":[{\"title\":\"Node\",\"value\":\"${NODE_NAME}\",\"short\":false},{\"title\":\"Instance\",\"value\":\"${INSTANCE_ID}\",\"short\":true},{\"title\":\"Instance Type\",\"value\":\"${INSTANCE_TYPE}\",\"short\":true},{\"title\":\"Availability Zone\",\"value\":\"${AZ}\",\"short\":true},{\"title\":\"Surviving Worker Count\",\"value\":\"${SWC_MESSAGE}\",\"short\":true}]}]}" "${SLACK_URL}"
```
Comment (egeland) [inline entrypoint.sh]: Same here..

File: entrypoint.sh [hunk 3]
```diff
@@ -111,7 +127,7 @@ fi
 # - USA: https://event-receiver.sematext.com/APPLICATION_TOKEN/event
 # - EUROPE: https://event-receiver.sematext.com/APPLICATION_TOKEN/event
 if [ "${SEMATEXT_URL}" != "" ]; then
-  curl -X POST --data "{\"message\":\"${MESSAGE}\",\"title\":\"Spot Termination ${CLUSTER_INFO}\",\"host\":\"${NODE_NAME}\",\"Instance\":\"${INSTANCE_ID}\",\"Instance Type\":\"${INSTANCE_TYPE}\", \"Availability Zone\":\"${AZ}\", \"type\":\"aws_spot_instance_terminated\"}" "${SEMATEXT_URL}"
+  curl -X POST --data "{\"message\":\"${MESSAGE}\",\"title\":\"Spot Termination ${CLUSTER_INFO}\",\"host\":\"${NODE_NAME}\",\"Instance\":\"${INSTANCE_ID}\",\"Instance Type\":\"${INSTANCE_TYPE}\", \"Availability Zone\":\"${AZ}\", \"Surviving Instance Count\":\"${SWC_MESSAGE}\", \"type\":\"aws_spot_instance_terminated\"}" "${SEMATEXT_URL}"
```
Comment (egeland) [inline entrypoint.sh]: And here..

**Review by egeland — CHANGES_REQUESTED** [review]

File: entrypoint.sh (hunk 1 above)
Comment (egeland) [reply entrypoint.sh]: I definitely don't mind adding the feature, but we need to be clean in the use case when the feature is not being activated. :)

Comment (osellers) [reply entrypoint.sh]: Sorry for the delay, and thanks for the feedback. I plan on making the changes you suggest as soon as I find some time.

**Pull request closed without merging**